napari/utils/colormaps/colormap_utils.py

```diff
@@ -86,8 +86,10 @@ def convert_vispy_colormap(colormap, name='vispy'):
     # but if they do, we want to use it
     if hasattr(colormap, '_controls'):
         controls = colormap._controls
+        colors = colormap.colors.rgba
     else:
-        controls = np.zeros((0,))
+        controls = np.linspace(0, 1)
+        colors = np.array([colormap.map(x) for x in controls])
```

Comment:
In this place interpolation is for 50 points. If I good read vispy code all problematic colormaps are linear so it will be enough to use 2 points. But approaches used in vispy code allow implementing any nonlinear mapping. So I decide to use 50 points approximation.